openshift_tools/monitoring/generic_metric_sender.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# vim: expandtab:tabstop=4:shiftwidth=4
+"""
+Base class for sending metrics
+"""
+
+import os
+import yaml
+
+
+class GenericMetricSenderException(Exception):
+    '''
+        GenericMetricSenderException
+        Exists to propagate errors up from the api
+    '''
+    pass
+
+
+class GenericMetricSender(object):
+    """
+    Abstract class that encapsulates all metric sending operations.
+    """
+
+    def __init__(self):
+        self.config = None
+        self.config_file = None
+
+    def parse_config(self):
+        """ parse default config file """
+        if not self.config:
+            if not os.path.exists(self.config_file):
+                raise GenericMetricSenderException(self.config_file + " does not exist.")
+            self.config = yaml.load(file(self.config_file))
+
+    def add_dynamic_metric(self, discovery_key, macro_string, macro_array, host=None, synthetic=False):
+        pass
```

Comment:
@yaacov correct me if I'm wrong on this, but what we discussed before, is to have a separate file that would contain a key/value pair of item name and all tags that should be associated with it.

So IMO, a dynamic item would be that plus extra tags for items passed into the function.

Thoughts?